src/libproc/proc_pid.rs

```diff
@@ -281,7 +283,15 @@ pub fn pidpath(pid: i32) -> Result<String, String> {
 
 #[cfg(target_os = "linux")]
 pub fn pidpath(pid: i32) -> Result<String, String> {
-    unimplemented!()
+    let exe_path = CString::new(format!("/proc/{}/exe", pid)).unwrap();
```

Comment:
Question (I haven't looked into this yet):
Could you just use format!() and a String here, and then use .as_ptr() when you pass it to readlink()?
Do you need to create the CString at all?
You're doing it like this https://thefullsnack.com/en/string-ffi-rust.html so is probably correct and saves manually adding NULL and stuff like that.

Reply:
Yeah I used the CString since rust strings aren't null terminated. I can manually add the null if desired though

Reply:
No worries. If this is the recommended way to do it it's perfect.
I'll add some comment/doc changes on top of this and merge it